test/controllers/users_controller_test.rb

```diff
@@ -290,9 +290,21 @@ def change_user_password(user_factory, new_password)
           assert_equal @response.content_type, 'text/csv'
         end
 
-        should "export filtered users" do
+        should "export filtered users by role" do
           user = create(:user)
-          get :index, params: { role: 'admin', format: :csv }
+          get :index, params: { role: 'normal', format: :csv }
+          lines = @response.body.lines
+          assert_equal(2, lines.length)
+        end
+
+        should "export filtered users by all fields" do
+          organisation = create(:organisation, name: "Cookies Of Other Lands", abbreviation: 'COOL')
+
+          create(:suspended_user, name: "Special cookie", email: "specialcookie@email.com", role: 'normal', organisation: organisation)
+          create(:suspended_user, name: "Normal cookie", email: "normalcookie@email.com", role: 'normal', organisation: organisation)
+
+          get :index, params: { role: 'normal', status: 'suspended', organisation: organisation, two_step_status: false, filter: 'special', format: :csv }
+
           lines = @response.body.lines
           assert_equal(2, lines.length)
```

Comment:
When I first read this I wondered if any users were being filtered out, it's not that clear the signed in user is omitted from the export.
Perhaps before this assertion something like:
```ruby
assert_equal(3, User.count)
```
Would make this more obvious.

Reply:
That's a good point - thanks @steventux! I have added this in now 👍